### PR TITLE
Reuse docker image

### PR DESCRIPTION
**Pull request by troibe**
**What?**
This commit introduces a github action that takes the tool as an input and returns the latest version of its docker image.

**Why?**
We currently copy and paste the code to retrieve the latest version of the `sc_tools` images.
This is also the case for sclib, zerosoc and scscerver.

**Successful run:**
https://github.com/siliconcompiler/siliconcompiler/actions/runs/6039399625

### CODE REVIEW

File: .github/workflows/docker_image.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  workflow_call:
+    inputs:
+      tool:
+        description: 'Name of the tool or "tools" for all tools'
+        required: true
+        type: string
+    outputs:
+      sc_tool:
+        description: 'Name and newest digest for the docker image of the input tool'
+        value: ${{ jobs.docker_image.outputs.sc_tool }}
```
Comment (gadfort) [inline .github/workflows/docker_image.yml:16]: can you add an input for sc version as the repos can control that via a key?

Comment (troibe) [reply .github/workflows/docker_image.yml:17]: Sure added that now.
Didn't knwo the `builder.py` script gives you the latest version for each tool when using an older commit of sc as well.

File: .github/workflows/docker_image.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  workflow_call:
+    inputs:
+      tool:
+        description: 'Name of the tool'
+        required: false
+        default: 'tools'
+        type: string
+      sc_version:
+        description: 'SC Version tag'
+        required: false
```
Comment (gadfort) [inline .github/workflows/docker_image.yml:16]: does this need a default value of 'main'?

File: .github/workflows/docker_image.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+on:
+  workflow_call:
+    inputs:
+      tool:
+        description: 'Name of the tool'
+        required: false
+        default: 'tools'
+        type: string
+      sc_version:
+        description: 'SC Version tag'
```
Comment (gadfort) [inline .github/workflows/docker_image.yml:15]: SC version tag

File: .github/workflows/docker_image.yml (hunk 2 above)
Comment (troibe) [reply .github/workflows/docker_image.yml:16]: No runs successfully like that.

File: .github/workflows/docker_image.yml (hunk 3 above)
Comment (troibe) [reply .github/workflows/docker_image.yml]: Woops in

**Review by gadfort — APPROVED** [review]

**Pull request merged**